### DISCUSSION TITLE
Welcome to corrosion Discussions!

### DISCUSSION THREAD

**Discussion by jschwe** [corrosion-rs/corrosion#187] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jason-yau** [comment]
Wanted-help: When I add more than one Rust static library which is built with cxx crate, duplicated symbol errors occurred. I think it is because the cxx.cc source was compiled into the rust library. Then the libraries have same symbols which is defined in cxx.cc. Thus, when more than one such library is linked, the errors occur. How do solve this issue?
```bash
duplicate symbol '_cxxbridge1$exception' in:
    rust/libfoo.a(cxx-6f92fa0cc8784bc2.cxx.25a7702b-cgu.12.rcgu.o)
    rust/libbar.a(cxx-50d76a46afddca83.cxx.7a1aee28-cgu.12.rcgu.o)
duplicate symbol '_cxxbridge1$str$new' in:
    rust/libfoo.a(cxx-6f92fa0cc8784bc2.cxx.25a7702b-cgu.14.rcgu.o)
    rust/libbar.a(cxx-50d76a46afddca83.cxx.7a1aee28-cgu.14.rcgu.o)
duplicate symbol '_cxxbridge1$str$ptr' in:
    rust/libfoo.a(cxx-6f92fa0cc8784bc2.cxx.25a7702b-cgu.14.rcgu.o)
    rust/libbar.a(cxx-50d76a46afddca83.cxx.7a1aee28-cgu.14.rcgu.o)
duplicate symbol '_cxxbridge1$str$len' in:
    rust/libfoo.a(cxx-6f92fa0cc8784bc2.cxx.25a7702b-cgu.14.rcgu.o)
    rust/libbar.a(cxx-50d76a46afddca83.cxx.7a1aee28-cgu.14.rcgu.o)
duplicate symbol '_cxxbridge1$str$from' in:
    rust/libfoo.a(cxx-6f92fa0cc8784bc2.cxx.25a7702b-cgu.14.rcgu.o)
    rust/libbar.a(cxx-50d76a46afddca83.cxx.7a1aee28-cgu.14.rcgu.o)
    ........
```

**Reply by jschwe** [reply]
So just to clarify, you have:
- Some C++ library that you link into multiple Rust libraries.
- You then link those Rust libraries as static libraries into a foreign (e.g. C/C++) library

You should only invoke the linker once when creating the final binary. So, in your case, don't link your C++ code into the Rust library but defer the linking until the final step. You should be able to just do `target_link_libraries(<rust_library> INTERFACE <cxx_library>)` in CMake.
